dso/dependabot.py

```diff
@@ -0,0 +1,187 @@
+import datetime
+import pathlib
+import tabulate
+import tempfile
+import typing
+
+import ccc.github
+import ci.util
+import dso.model
+import dso.util
+import gitutil
+
+
+def _dependabot_yaml_in_known_places(repo_dir: str) -> bool:
+    clogger = dso.util.component_logger(__name__)
+    DEPENDABOT_YAML_PATHS = (
+        '.dependabot.yaml',
+        '.dependabot.yml',
+        '.github/.dependabot.yaml',
+        '.github/.dependabot.yml',
+        'dependabot.yaml',
+        'dependabot.yml',
+        '.github/dependabot.yaml',
+        '.github/dependabot.yml'
+    )
+
+    repo_dir = ci.util.existing_dir(pathlib.Path(repo_dir))
+    if not repo_dir.joinpath('.git').is_dir():
+        raise ValueError(f'not a git root directory: {repo_dir}')
+
+    for path in DEPENDABOT_YAML_PATHS:
+        dependabot_yaml = repo_dir.joinpath(path)
+        if dependabot_yaml.is_file():
+            return True
+
+    clogger.info('dependabot not found in any known places')
+    return False
+
+
+def is_dependabot_in_component_repo(
+    repo: str,
+    github: str,
+    github_cfg
+) -> bool:
+    clogger = dso.util.component_logger(__name__)
+
+    clogger.info(f'cloning {github}/{repo}')
+    with tempfile.TemporaryDirectory() as temp_dir:
+        gitutil.GitHelper.clone_into(
+            target_directory=temp_dir,
+            github_cfg=github_cfg,
+            github_repo_path=repo,
+        )
+
+        return _dependabot_yaml_in_known_places(repo_dir=temp_dir)
+
+
+def _get_repo_in_org(org_name: str):
+    clogger = dso.util.component_logger(__name__)
+    host, org = org_name.split('/')
+    clogger.info(f'looking for repos from {org=} on {host=}')
+
+    github_cfg = ccc.github.github_cfg_for_hostname(
+        cfg_factory=ci.util.ctx().cfg_factory(),
+        host_name=host,
+    )
+    github_api = ccc.github.github_api(github_cfg)
+    github_org = github_api.organization(org)
+    return github_org.repositories
+
+
+def _scan_repos_for_dependabot(
+    repos: typing.List[str],
+    host: str,
+) -> dso.model.DependabotCoverageReport:
+
+    clogger = dso.util.component_logger(__name__)
+    i = 1
+
+    report: dso.model.DependabotCoverageReport
+    details = []
+
+    clogger.info(f'scanning {len(repos)} repos')
+    for repo in repos:
+        github_cfg = ccc.github.github_cfg_for_hostname(
+            cfg_factory=ci.util.ctx().cfg_factory(),
+            host_name=host,
+        )
+        clogger.info(f'{i} / {len(repos)}')
+        dependabot_found = is_dependabot_in_component_repo(
+            repo=repo,
+            github_cfg=github_cfg,
+            github=host
+        )
+        repo_report = dso.model.DependabotCoverageReportRepo(
+            dependabot=dependabot_found,
+            repo=repo
+        )
+        details.append(repo_report)
+        i += 1
+
+    return dso.model.DependabotCoverageReport(
+        coverage=0,
+        details=details,
+        github=host,
+    )
+
+
+def _get_github_api(host):
```

Comment:
nit: I would prefer to not name functions that are not simple getters (as in: return an attr (from an object), maybe doing some trivial operations on the attr) with a `get` prefix.

regardless if my nit, I think this convenience function should be moved to `ccc.github`, where it is obviously missing (in fact, I also needed such a function several times, but always was too lazy to implement one..)

+ please add a typehint for the return-value (and preferrably also the parameter - even though the latter is debatable, as it is just `str`..)